alpha/apps/kaltura/modules/extwidget/actions/embedPlaykitJsAction.class.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/**
+ * @package Core
+ * @subpackage externalWidgets
+ */
+class embedPlaykitJsAction extends sfAction
+{
+    var $eTagHash = null;
+    var $partner_id = null;
+    var $ui_conf_id = null;
+    var $sourcesPath = null;
+    var $bundle_name = null;
+
+    public function execute()
+    {
+        //Get uiConf ID from QS
+        $this->uiconf_id = $this->getRequestParameter('uiconf_id');
+        if (!$this->uiconf_id)
+            KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, 'uiconf_id');
+
+        // retrieve uiCong Obj
+        $uiConf = uiConfPeer::retrieveByPK($this->uiconf_id);
+        if (!$uiConf)
+            KExternalErrors::dieError(KExternalErrors::UI_CONF_NOT_FOUND);
+
+        //Get partner ID from QS
+        $this->partner_id = $this->getRequestParameter('partner_id', $uiConf->getPartnerId());
+        if (!$this->partner_id)
+            KExternalErrors::dieError(KExternalErrors::MISSING_PARAMETER, 'partner_id');
+
+        $bundleWebDirPath = kConf::get('playkit_js_bundles_path');
```

Comment:
If you are using an object , then better move all the kConf get to the constructor and also validate their existance